src/hooks/useUpdateDeployedContractDetails.ts

```diff
@@ -32,10 +34,67 @@ export const useUpdateDeployedContractDetails = () => {
           deployedContractDetails: response.data
         });
 
+        if (
+          response.data.isVerified &&
+          (!smartContract.rawAbi ||
+            smartContract.rawAbi?.isFromVerifiedContract)
+        ) {
+          const url = `accounts/${address}/verification`;
+          const response = await get({ url });
+          if (response?.success && response?.data?.source?.abi) {
+            const verifiedContractAbi = response.data.source.abi;
+
+            try {
+              if (!verifiedContractAbi.name) {
+                verifiedContractAbi.name = INTERFACE_NAME_PLACEHOLDER;
+              }
+              smartContractDispatch({
+                type: SmartContractDispatchTypeEnum.setVerifiedContract,
+                verifiedContract: response.data
+              });
+              smartContractDispatch({
+                type: SmartContractDispatchTypeEnum.setRawAbi,
+                rawAbi: { ...verifiedContractAbi, isFromVerifiedContract: true }
+              });
+
+              const abiRegistry = getAbiRegistry(verifiedContractAbi);
+              smartContractDispatch({
+                type: SmartContractDispatchTypeEnum.setAbiRegistry,
+                abiRegistry
+              });
+            } catch (error) {
+              console.error('Unable to parse Verified Contract ABI: ', error);
+            }
+            setIsContractAddressCheckLoading(false);
+
+            return;
+          }
+        }
+
+        smartContractDispatch({
+          type: SmartContractDispatchTypeEnum.setVerifiedContract,
+          verifiedContract: undefined
+        });
+        if (smartContract.rawAbi?.isFromVerifiedContract) {
```

Comment:
there are some early returns on the "happy path", otherwise depending on the circumstances, the code should reach the dispatches at the end to "reset" the existing config